scripts/crc-deploy.sh

```diff
@@ -0,0 +1,193 @@
+  image:
+    ref: OLM_IMAGE
+package:
+  image:
+    ref: OLM_IMAGE
+EOF
+
+  sed "${SED_OPTS[@]}" "${VALUES_TEMPLATE}" > "${CRC_E2E_VALUES}"
+
+  # psm operator patch
+  PSM_OPERATOR_PATCH_TEMPLATE=$(mktemp)
+  cat << EOF > "${PSM_OPERATOR_PATCH_TEMPLATE}"
+- command: update
+  path: spec.template.spec.containers[0].image
+  value: OLM_IMAGE
+- command: update
+  path: spec.template.spec.containers[0].env[1].value
+  value: OLM_IMAGE
+EOF
+
+  sed "${SED_OPTS[@]}" "${PSM_OPERATOR_PATCH_TEMPLATE}" > scripts/psm-operator-deployment.crc.e2e.patch.yaml
+
+  # collect-profiles patch
+  COLLECT_PROFILE_PATCH_TEMPLATE=$(mktemp)
+  cat << EOF > "${COLLECT_PROFILE_PATCH_TEMPLATE}"
+- command: update
+  path: spec.jobTemplate.spec.template.spec.containers[0].image
+  value: OLM_IMAGE
+EOF
+
+  sed "${SED_OPTS[@]}" "${COLLECT_PROFILE_PATCH_TEMPLATE}" > scripts/collect-profiles.crc.e2e.patch.yaml
+}
+
+# YQ for applying yaml patches
+YQ="go run ./vendor/github.com/mikefarah/yq/v3/"
+
+# CRC_E2E_VALUES is the name of the help values file
+# used to populate manifest templates with the locally built olm images
+export CRC_E2E_VALUES="values-crc-e2e.yaml"
+
+# Set kubeconfig to CRC if necessary
+export KUBECONFIG=${KUBECONFIG:-${HOME}/.crc/machines/crc/kubeconfig}
+KUBE_ADMIN_USER=$(crc console --credentials -o json | jq -r .clusterConfig.adminCredentials.username)
+KUBE_ADMIN_PASSWORD=$(crc console --credentials -o json | jq -r .clusterConfig.adminCredentials.password)
+
+# login to crc
+echo "Logging in as kubeadmin"
+oc login -u "${KUBE_ADMIN_USER}" -p "${KUBE_ADMIN_PASSWORD}" > /dev/null
+
+# Scale down CVO to stop changes from returning to stock configuration
+echo "Scaling down CVO"
+oc scale --replicas 0 -n openshift-cluster-version deployments/cluster-version-operator
+
+echo "Allow cluster wide access to openshift repository"
+oc policy add-role-to-group system:image-puller system:serviceaccounts --namespace=openshift
+
+# push olm images to crc global repository
+push_images olm:test opm:test
+
+SKIP_MANIFESTS=${SKIP_MANIFESTS:-0}
+if [ "${SKIP_MANIFESTS}" = 0 ]; then
+  # Create values and patches files
+  # Get images with the specific shas
+  OLM_IMG="$(oc get istag/olm:latest -o json | jq -r .image.dockerImageReference)"
+  OPM_IMG="$(oc get istag/opm:latest -o json | jq -r .image.dockerImageReference)"
+
+  make_manifest_patches "${OLM_IMG}" "${OPM_IMG}"
+
+  # Build e2e manifests
+  echo "Generating manifests"
+  # CRC_E2E_VALUES is already set in this script and exported
+  # If set, it will include the file referenced by it in the helm template command
+  # and update the manifests to use the locally built images
+  ./scripts/generate_crds_manifests.sh
+
+  # Apply patches
+  ${YQ} write --inplace -s scripts/psm-operator-deployment.crc.e2e.patch.yaml manifests/0000_50_olm_06-psm-operator.deployment.yaml
+  ${YQ} write --inplace -s scripts/collect-profiles.crc.e2e.patch.yaml manifests/0000_50_olm_07-collect-profiles.cronjob.yaml
```

Comment:
Ran into another error while running this locally and manually setting the current kubecontext to crc-admin:

```console
$ make crc-deploy
...
wrote ./tmp.hxG09zV0mc/chart/olm/templates/0000_50_olm_13-operatorgroup-default.yaml
wrote ./tmp.hxG09zV0mc/chart/olm/templates/0000_50_olm_13-operatorgroup-default.yaml
wrote ./tmp.hxG09zV0mc/chart/olm/templates/0000_90_olm_01-prometheus-rule.yaml
wrote ./tmp.hxG09zV0mc/chart/olm/templates/0000_90_olm_00-service-monitor.yaml
wrote ./tmp.hxG09zV0mc/chart/olm/templates/0000_90_olm_00-service-monitor.yaml

go run: no packages loaded from ./vendor/github.com/mikefarah/yq/v3/
make: *** [Makefile:180: crc-deploy] Error 1
```